consensus/src/epoch_manager.rs

```diff
@@ -1435,6 +1434,70 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
                 self.config.quorum_store.batch_expiry_gap_when_init_usecs;
             let payload_manager = self.payload_manager.clone();
             let pending_blocks = self.pending_blocks.clone();
+            let perform_pessimistic_verification = self
+                .epoch_state()
+                .verifier
+                .pessimistic_verify_set()
+                .contains(&peer_id);
+
+            if self.config.optimistic_sig_verification_for_order_votes
+                && !perform_pessimistic_verification
+            {
+                if let UnverifiedEvent::OrderVoteMsg(order_vote) = &unverified_event {
+                    order_vote.verify_metadata()?;
```

Comment:
it makes more sense to make no changes here but having optimistic_verify() inside the verify function.
also it doesn't make much sense to have separate bools to control where it's enabled for one message type or not, we should just have one bool to control the whole feature.

basically all the verify function will look like
```
fn verify() {
  verify_metadata()?;
  verifier.optimistic_verify(, &signature_with_status)?
}
```
```
fn optimistic_verify(..) -> Result<bool> {
  if enabled {
    if !pessimistic {
      return Ok(false); // skipped
    }
  }
  verify(..).map_err(OptimisticVerifyResult::Failed)?;
  signature.mark_verified();
  Ok(true) // verified
}
```

Reply:
Done.

Reply:
it doesn't look like what I suggested, I think the `optimistic_verify` should be moved into the ValidatorVerify itself, so all the caller can just re-use the function without any changes. when construct the ValidatorVerifier it can take a bool `enable_optimistic_verify`

Reply:
Yes, this wasn't exactly what you mentioned. I moved the verification logic out of epoch manager. But I didn't move the verification logic to validator verifier. I felt the validator verifier offers a more lower-level functionality that's used in many places. So, I wasn't sure if it's semantically a good choice to push this optimistic signature verification of votes from vote.rs to valdiator verifier. If you really prefer that way, I can move the logic to validator verifier.

Reply:
I imagine with proper api, the whole optimistic verify logic can be self-contained in the validator verifier.
```
fn optimistic_verify()
fn aggregate_and_verify()
fn aggregate_verified_signatures()
```
probably the ledger info with unverified signatures should just be a wrapper around it.